CoEpi/domain/logic/CenLogic.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import CryptoSwift
+
+class CenLogic {
+    private let CENKeyLifetimeInSeconds: Int64 = 2 * 60 // TODO: revert back to 7*86400
+    private let CENLifetimeInSeconds: Int64 = 1 * 60 // TODO: revert back to 15*60
+
+    func shouldGenerateNewCenKey(curTimestamp: Int64, cenKeyTimestamp: Int64) -> Bool {
+         (cenKeyTimestamp == 0) || (roundedTimestamp(ts: curTimestamp) > roundedTimestamp(ts: cenKeyTimestamp))
+    }
+
+    func generateCenKey(curTimestamp: Int64) -> CENKey {
+        //generate a new AES Key and store it in local storage
+
+        //generate base64string representation of key
+        let cenKeyString = computeSymmetricKey()
+        let cenKeyTimestamp = curTimestamp
+
+        //Create CENKey and insert/save to Realm
+        let newCENKey = CENKey(cenKey: cenKeyString, timestamp: cenKeyTimestamp)
+        
+        return newCENKey
+    }
+
+    // TODO ideally this should return Cen, not Data. Implement Cen <-> Data (separately)
+    func generateCen(CENKey: String) -> Data {
+        let currentTs : Int64 = Int64(Date().timeIntervalSince1970)
+        // decode the base64 encoded key
+        let decodedCENKey:Data = Data(base64Encoded: CENKey)!
+
+        //convert key to [UInt8]
+        var decodedCENKeyAsUInt8Array: [UInt8] = []
+        decodedCENKey.withUnsafeBytes {
+            decodedCENKeyAsUInt8Array.append(contentsOf: $0)
+        }
+
+        //convert timestamp to [UInt8]
+        var tsAsUInt8Array: [UInt8] = []
+        [roundedTimestamp(ts: currentTs)].withUnsafeBytes {
+            tsAsUInt8Array.append(contentsOf: $0)
+        }
+
+        //encrypt tsAsUnit8Array using decodedCENKey... using AES
+        let encData = try! AES(key: decodedCENKeyAsUInt8Array, blockMode: ECB(), padding: .pkcs5).encrypt(tsAsUInt8Array)
```

Comment:
Could reduce all of this to a computed var  on CENKey, or have a toData on CEN

Reply:
CenLogic could be part of CenKey class, but this also works.

Reply:
That could work, though what if we wanted to try out different generation algorithms? We wouldn't be able to swap those easily.

I'd perhaps consider splitting this class in `CenKeyGenerator` and `CenGenerator`?